torch/backends/_nnapi/serializer.py

```diff
@@ -0,0 +1,1131 @@
+import enum
+import collections
+import struct
+import operator
+import functools
+import logging
+from typing import (
+        Tuple,
+        NamedTuple,
+)
+
+import torch
+
+
+
+#TODO: Add type annotations
+#TODO: Check tensor types for ops
+
+
+LOG = logging.getLogger("nnapi_serialize")
+
+
+class NNAPI_OperandCode(object):
+    FLOAT32 = 0
+    INT32 = 1
+    UINT32 = 2
+    TENSOR_FLOAT32 = 3
+    TENSOR_INT32 = 4
+    TENSOR_QUANT8_ASYMM = 5
+    BOOL = 6
+    TENSOR_QUANT16_SYMM = 7
+    TENSOR_FLOAT16 = 8
+    TENSOR_BOOL8 = 9
+    FLOAT16 = 10
+    TENSOR_QUANT8_SYMM_PER_CHANNEL = 11
+    TENSOR_QUANT16_ASYMM = 12
+
+
+class NNAPI_OperationCode(object):
```

Comment:
Each NNAPI op defines its own semantics.  For example, Conv2D has two forms, one with 10 args and one with 13: https://android.googlesource.com/platform/frameworks/ml/+/refs/tags/android-11.0.0_r8/nn/runtime/include/NeuralNetworks.h#401 .  But if you look closer, some of those args are "Available since API level 29", so there are also a 7-arg version and a *different* 10-arg version.  I believe the two different 10-arg versions are distinguished by the types of the arguments.

Reply:
I see. We may need to find the bast schema match of Conv2D manually, when converting a conv2d node to nnapi::Conv2D